api-helpers/api-hofs.ts

```diff
@@ -105,3 +102,29 @@ export function withAuth<R extends NextApiRequest>(role?: UserRole) {
     return handler({ session, ...req }, res);
   };
 }
+
+type HTTPMethod =
+  | 'get'
+  | 'head'
+  | 'post'
+  | 'put'
+  | 'delete'
+  | 'connect'
+  | 'options'
+  | 'trace'
+  | 'patch';
+
+export function withMethods<R extends NextApiRequest>(
```

Comment:
Obawiam się że to nie zadziała za dobrze jeśli chodzi o typy, ale musiałbym sprawdzić